dev/deps/spark-deps-hadoop-2.7

```diff
@@ -17,13 +18,17 @@ arpack_combined_all-0.1.jar
 arrow-format-0.12.0.jar
 arrow-memory-0.12.0.jar
 arrow-vector-0.12.0.jar
+ast-9.0-9.0.20190305.jar
 automaton-1.11-8.jar
 avro-1.8.2.jar
 avro-ipc-1.8.2.jar
 avro-mapred-1.8.2-hadoop2.jar
 bonecp-0.8.0.RELEASE.jar
 breeze-macros_2.12-0.13.2.jar
 breeze_2.12-0.13.2.jar
+cats-core_2.12-1.6.0.jar
+cats-kernel_2.12-1.6.0.jar
+cats-macros_2.12-1.6.0.jar
```

Comment:
@s1ck . This seems to introduce many transitive dependencies. I'm wondering if we can reduce these transitive dependency by adding exclusion rules?

cc @mengxr

Reply:
Those dependencies are part of the OKAPI stack which is responsible for translating Cypher to relational operations. I understand that adding a lot of dependencies is bad practice for a single PR. By excluding them, we basically require the user to add explicit dependencies? If yes, it doesn't sound like an elegant solution to me.

Reply:
@s1ck . The alternative in my mind is making these into `external` modules like `external/avro` and `external/kafka-0-10`.

Hi, @rxin , @mengxr , @gatorsmile . Can we make new Graph modules as `external` modules instead? This looks too intrusive to me.

Reply:
@dongjoon-hyun This module adds a significant feature in a major Spark release. Is this not a reason to have a more relaxed qualification on dependent libraries?

Reply:
KAFKA and AVRO also does.